app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt

```diff
@@ -0,0 +1,180 @@
+                    .fillMaxWidth(),
+            )
+        }
+    }
+}
+
+@Composable
+private fun InformationSection(
+    crossSellingResult: CrossSellingResult,
+    modifier: Modifier
+) {
+    @DrawableRes
+    val drawableId: Int = when (crossSellingResult) {
+        is CrossSellingResult.Success -> R.drawable.ic_checkmark_in_circle
+        CrossSellingResult.Error -> R.drawable.ic_x_in_circle
+    }
+    val titleText = when (crossSellingResult) {
+        CrossSellingResult.Error -> "Something went wrong."
+        is CrossSellingResult.Success -> {
+            stringResource(
+                R.string.purchase_confirmation_new_insurance_today_app_state_title,
+                crossSellingResult.insuranceType
+            )
+        }
+    }
+    val descriptionText = when (crossSellingResult) {
+        CrossSellingResult.Error -> "Your purchase couldn't be completed.\nContact us in the chat."
+        is CrossSellingResult.Success -> {
+            when {
+                crossSellingResult.startingDate.dayOfMonth <= LocalDate.now().dayOfMonth -> {
+                    stringResource(R.string.purchase_confirmation_new_insurance_today_app_state_description)
+                }
+                else -> {
+                    val activationDate = crossSellingResult.startingDate.format(
+                        DateTimeFormatter.ISO_LOCAL_DATE
+                    )
+                    stringResource(
+                        R.string.purchase_confirmation_new_insurance_active_in_future_app_state_description,
+                        activationDate
+                    )
+                }
+            }
+        }
+    }
+    Column(
+        modifier = modifier,
+        verticalArrangement = Arrangement.spacedBy(18.dp),
+    ) {
+        Image(
+            painter = painterResource(drawableId),
+            contentDescription = null,
+            modifier = Modifier.size(32.dp),
+        )
+        Text(
+            titleText,
+            style = MaterialTheme.typography.h5
+        )
+        Text(
+            descriptionText,
+            style = MaterialTheme.typography.body2
+        )
+    }
+}
+
+@Composable
+private fun ButtonsSection(
+    crossSellingResult: CrossSellingResult,
+    modifier: Modifier,
+) {
+    Column(
+        modifier = modifier,
+        verticalArrangement = Arrangement.spacedBy(16.dp)
+    ) {
+        when (crossSellingResult) {
+            is CrossSellingResult.Error -> {
+                LargeContainedButton(
+                    onClick = {},
+                    modifier = Modifier.fillMaxWidth()
+                ) {
+                    Image(
+                        painter = painterResource(R.drawable.ic_chat_white),
+                        contentDescription = null,
+                    )
+                    Spacer(Modifier.width(8.dp))
+                    Text("Open chat")
+                }
+                LargeOutlinedButton(
+                    onClick = {},
```

Comment:
I think this needs to be wired up to something 😉

Reply:
Absolutely, but this would be part of "wiring it up to the rest of the app" which I'd need some help on. Since right now there's no "normal" way to go into this screen anyway